src/FormLabel.js

```diff
@@ -59,20 +59,20 @@ const FormLabel = React.forwardRef(
       column && 'col-form-label',
     );
 
-    if (column) return <Col {...props} className={classes} as="label" />;
-
     warning(
       controlId == null || !htmlFor,
       '`controlId` is ignored on `<FormLabel>` when `htmlFor` is specified.',
     );
+    htmlFor = htmlFor || controlId;
+
+    if (column)
+      return (
+        <Col {...props} htmlFor={htmlFor} className={classes} as="label" />
```

Comment:
```suggestion
        <Col as="label" className={classes} htmlFor={htmlFor} {...props} />
```
enhance logical order of prop types a bit :)

Reply:
It goes a bit beyond a simple defect fix this PR is supposed to be but ok, do it rather now than never :-)